### PR TITLE
ISAICP-5017 - Show update date on Document, News, ...

### PR DESCRIPTION
**Pull request by Joosthe**

### CODE REVIEW

File: web/themes/joinup/joinup_theme.theme [hunk 1]
```diff
@@ -1048,13 +1054,23 @@ function joinup_theme_preprocess_page_title__node__document(&$variables) {
...
+  $date_formatter = \Drupal::service('date.formatter');
+  $field_document_publication_date = $view_builder->viewField($node->field_document_publication_date, 'default');
   $variables['additional_content']['field_document_publication_date'] = $field_document_publication_date;
+  $publication_date_value = $node->field_document_publication_date->value;
+  // The raw value allowed to properly display the field above with any class
+  // or configuration applied but still have the value to compare to the changed
+  // date in the twig.
+  $variables['additional_content']['field_document_publication_date_raw'] = $date_formatter->format(strtotime($publication_date_value), 'custom', 'd/m/Y');
+
+  $updateDate = $date_formatter->format($node->getChangedTime(), 'custom', 'd/m/Y');
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply web/themes/joinup/joinup_theme.theme]: I agree. This was copied over from another wrongly defined variable.

File: web/modules/custom/joinup_community_content/joinup_community_content.behat.inc [hunk 2]
```diff
@@ -191,4 +191,34 @@ class JoinupCommunityContentSubContext extends DrupalSubContextBase implements D
...
+   *   The node label.
+   * @param string $new_date
+   *   The new date property.
+   *
+   * @Given the :field_name date of the :title :bundle is :new_date
+   */
+  public function updateNodeDateProperty(string $field_name, string $title, string $bundle, string $new_date): void {
+    $node = $this->getNodeByTitle($title, $bundle);
+    if (!($time = strtotime($new_date))) {
+      throw new Exception("{$new_date} could not be converted to string.");
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply web/modules/custom/joinup_community_content/joinup_community_content.behat.inc]: Meh, you have noted this too many times and I still forget it.. I wonder why this is not in the code sniffing issues!

File: web/modules/custom/joinup_community_content/joinup_community_content.behat.inc [hunk 3]
```diff
@@ -212,6 +215,10 @@ class JoinupCommunityContentSubContext extends DrupalSubContextBase implements D
     }
     $node->set($field_name, $time);
     $node->save();
+
+    // Since this is a direct node update, the cache tags have to be invalidated
+    // manually for the pages to be updated.
+    \Drupal::service('cache_tags.invalidator')->invalidateTags($node->getCacheTagsToInvalidate());
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply web/modules/custom/joinup_community_content/joinup_community_content.behat.inc]: Indeed it works without it.. I guess I was manually refreshing the page while the test was running and caused the cache to work wrongly. I ran it now without issues.

File: web/modules/custom/joinup_community_content/joinup_community_content.behat.inc (hunk 2 above)
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply web/modules/custom/joinup_community_content/joinup_community_content.behat.inc]: This is stated here https://www.drupal.org/docs/develop/coding-standards/namespaces#s-use-ing-classes:

> Classes and interfaces without a backslash \ inside their fully-qualified name (for example, the built-in PHP Exception class) must be fully qualified when used in a namespaced file. For example: new \Exception();. Do not use global classes.

But as I see it's still not fixed :)

File: web/modules/custom/joinup_community_content/joinup_community_content.behat.inc [hunk 4]
```diff
@@ -191,4 +191,34 @@ class JoinupCommunityContentSubContext extends DrupalSubContextBase implements D
     $this->visitPath($path);
   }
 
+  /**
+   * Update one of the date properties of a node and clear the cache.
```
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply web/modules/custom/joinup_community_content/joinup_community_content.behat.inc]: s/Update/Updates still not fixed.

File: web/themes/joinup/joinup_theme.theme [hunk 5]
```diff
@@ -1009,6 +1009,9 @@ function joinup_theme_preprocess_page_title__node__news(&$variables) {
   $published = \Drupal::service('date.formatter')->format($node->published_at->published_at_or_created, 'date_only');
   $variables['additional_content']['published'] = $published;
 
+  $updateDate = \Drupal::service('date.formatter')->format($node->getChangedTime(), 'custom', 'd/m/Y');
+  $variables['additional_content']['updated'] = $updateDate;
```
Comment (claudiu-cristea) [inline web/themes/joinup/joinup_theme.theme]: Still have mixed lower camel case with snake case.

File: web/themes/joinup/joinup_theme.theme [hunk 6]
```diff
@@ -1023,6 +1026,9 @@ function joinup_theme_preprocess_page_title__node__discussion(&$variables) {
...
```
Comment (claudiu-cristea) [inline web/themes/joinup/joinup_theme.theme]: Same.

File: web/modules/custom/joinup_community_content/joinup_community_content.behat.inc (hunk 2 above)
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply web/modules/custom/joinup_community_content/joinup_community_content.behat.inc]: I've opened https://www.drupal.org/project/coder/issues/3120001

**Pull request merged**